taxcalc/functions.py

```diff
@@ -207,146 +207,107 @@ def AGI(_ymod1, c02500, c02700, e02615, c02900, XTOT,
+        floor as a decimal fraction of AGI
 
-        ID_Charity_crt_Cash : Deduction for charitable cash contributions;
-        ceiling as a percent of AGI
+        ID_Charity_crt_all : Deduction for all charitable contributions;
+        ceiling as a decimal fraction of AGI
 
-        ID_Charity_crt_Asset : Deduction for charitable asset contributions;
-        ceiling as a percent of AGI
+        ID_Charity_crt_noncash : Deduction for noncash charitable
+        contributions; ceiling as a decimal fraction of AGI
 
         ID_Charity_frt : Deduction for charitable contributions;
-        floor as a percent of AGI
+        floor as a decimal fraction of AGI
 
     Taxpayer Characteristics:
         e17500 : Medical expense
 
-        e18425 : Income taxes
-
-        e18450 : General Sales Tax
-
-        e18500 : Real Estate tax
+        e18400 : State and local taxes
 
-        e19200 : Total interest deduction
+        e18500 : Real-estate taxes
 
-        e19800 : Cash Contribution
+        e19200 : Interest paid
 
-        e19550 : Qualified Mortgage Insurance Premiums
+        e19800 : Charity cash contributions
 
-        e20100 : Charity non-cash contribution
+        e20100 : Charity noncash contributions
 
-        e20400 : Total Miscellaneous expense
-
-        e20550 : Unreimbursed employee business Expense
-
-        e20600 : Tax preparation fee
+        e20400 : Total Miscellaneous expenses
 
     Intermediate Variables:
         _posagi: positive AGI
 
     Returns
     -------
     c04470 : Itemized deduction amount
-
-    Warning
-    -------
-    Any additional keyword args, such as 'puf=True' here, must be
-    passed to the function at the END of the argument list. If you stick the
-    argument somewhere in the middle of the signature, you will get errors.
     """
     # Medical
     c17750 = ID_Medical_frt * _posagi
-    c17000 = max(0., e17500 - c17750)
-    # State and Local Income Tax, or Sales Tax
-    _statax = (1 - ID_StateLocalTax_HC) * max(e18400, 0.)
-    # Other Taxes (including state and local)
-    real_estate = (1 - ID_RealEstate_HC) * e18500
-    c18300 = _statax + real_estate + e18800 + e18900
+    c17000 = max(0., e17500 - c17750) * (1. - ID_Medical_HC)
+    # State and local taxes
+    c18300 = ((1. - ID_StateLocalTax_HC) * max(e18400, 0.) +
+              (1. - ID_RealEstate_HC) * e18500)
+    # Interest paid
+    c19200 = e19200 * (1. - ID_InterestPaid_HC)
+    # Charity
+    lim30 = min(ID_Charity_crt_noncash * _posagi, e20100)
+    c19700 = min(ID_Charity_crt_all * _posagi, lim30 + e19800)
+    charity_floor = ID_Charity_frt * _posagi  # floor is zero in present law
+    c19700 = max(0., c19700 - charity_floor) * (1. - ID_Charity_HC)
     # Casualty
-    if e20500 > 0:
-        c37703 = e20500 + ID_Casualty_frt * _posagi
-    else:
+    if e20500 > 0.0:  # assume e20500 was subject to a 10% disregard
+        c37703 = e20500 + 0.10 * _posagi  # add back disregarded amount
```

Comment:
Everything else look good to me, except this part. 

I notice that this "10% assumption" is probably stemmed from the fact that parameter `ID_Casualty_frt ` being `0.1`. That is, if this parameter were changed from `0.1` to say `0.2`, and everything else remain the same. 
Then this will become a "20% assumption", instead of "10% assumption", right? 
If so, should we replace the hardcoded value `0.10` with parameter `ID_Casualty_frt `?

cc @MattHJensen